### PR TITLE
Feature authentication

### PR DESCRIPTION
**Pull request by RasmusAChr**
# Added identity
- Added identity package for the project.
- Updated ChirpDBContext to inherit from IdentityDbContext to accomodate the new identity tables.
- New migration is created and applied to the database to accomodate identity tables.
- A principal (ApplicationUser) was created to be accomodate future claims.
- Created navbar template as a partial element and added it to the layout partial.
- Scaffolded identity login, register and logout page for future customization.

### CODE REVIEW

**Review by Tipskind — APPROVED** [review]
Areas/Identity/pages is structured well, and is open for more functionality.

File: src/Chirp.Web/Areas/Identity/Pages/Account/Register.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+            ///     directly from your code. This API may change or be removed in future releases.
+            /// </summary>
+            [Required]
+            [EmailAddress]
+            [Display(Name = "Email")]
+            public string Email { get; set; }
+            
+            
+            // For registering usernames
```
Comment (DragonFlyersx) [inline src/Chirp.Web/Areas/Identity/Pages/Account/Register.cshtml.cs:84]: Might want to add this at a later date.

**Review by Nikolaj787 — APPROVED** [review]
Changes look great and great to see that we have Identity user login now. Let's add some testing to it and maybe also fix the small unnecessary blank line CodeFactor suggested to remove

**Review by DragonFlyersx — COMMENTED** [review]
Why do we keep old migrations laying around from and old database?

File: src/Chirp.Web/Areas/Identity/Pages/Account/Login.cshtml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+﻿@page
+@model LoginModel
```
Comment (niko391a) [inline src/Chirp.Web/Areas/Identity/Pages/Account/Login.cshtml:2]: Maybe we should consider using the internal LoginModel, such that we have more control, of when to implement eventual package updates.

**Review by DragonFlyersx — APPROVED** [review]
Looks good might need to create an issue for creating user and making their own usernames, so that we can implement it in the future, since the skeleton is here

**Review by niko391a — APPROVED** [review]

**Comment by RasmusAChr** [comment]
> Why do we keep old migrations laying around from and old database?

We keep the old migrations to rollback to older database schemas. I.e. we currently have applied the new migration with identity functionality. If we wanted to, we could roll back to our initial database migration scheme. However then we wouldnt be able to use the identity functionality and we would lose the data in the newly created tables :)

**Pull request merged**